### PR TITLE
fix undistortion black pixel bug

### PR DESCRIPTION
**Pull request by homee-dennis**
Remove all black pixel after undistortion by referencing to [colmap implementation](https://github.com/colmap/colmap/blame/189478b69bf09b80b6143c491f5b29023ef73e7a/src/colmap/image/undistortion.cc#L777C13-L777C13).

### CODE REVIEW

File: arkit_utils/undistort_images/undistort_image.py [hunk 1]
```diff
@@ -4,6 +4,17 @@
 import os
 import argparse
 from concurrent.futures import ProcessPoolExecutor
+import math
+
+global undistort_data
+undistort_data = None
+
+class UndistortData:
+    def __init__(self, crop_left, crop_top, crop_width, crop_height):
```
Comment (homee-james) [inline arkit_utils/undistort_images/undistort_image.py]: i think it'd be better to add type annotation

File: arkit_utils/undistort_images/undistort_image_cuda.py [hunk 2]
```diff
@@ -5,6 +5,19 @@
 import argparse
 from numba import cuda
 import tqdm
+import math
+
+global undistort_data
+undistort_data = None
+
+class UndistortData:
+    def __init__(self, crop_left, crop_top, crop_width, crop_height):
```
Comment (homee-james) [inline arkit_utils/undistort_images/undistort_image_cuda.py]: duplicate code.

**Comment by homee-james** [comment]
I think overall the code can work and is well readable. However, I find there is high similarity between cuda version and cpu version, maybe we can merge the two files into one?

File: arkit_utils/undistort_images/undistort_image_cuda.py (hunk 2 above)
Comment (homee-dennis) [reply arkit_utils/undistort_images/undistort_image_cuda.py]: merge code in [this commit](https://github.com/homee-ai/nerfstudio/pull/9/commits/027bae0be48e8acc9f4b12e6bb9e798105443020)

File: arkit_utils/undistort_images/undistort_image.py (hunk 1 above)
Comment (homee-dennis) [reply arkit_utils/undistort_images/undistort_image.py]: added in [this commit](https://github.com/homee-ai/nerfstudio/pull/9/commits/6bdaac87ae1e101e844e3775af68d8f986dee624)

**Comment by homee-dennis** [comment]
Merge cpu and gpu version in latest commit

**Pull request merged**